test/framework/src/main/java/org/elasticsearch/node/MockNode.java

```diff
@@ -97,6 +100,21 @@ protected TransportService newTransportService(Settings settings, Transport tran
     }
 
     @Override
+    protected ZenPing newZenPing(Settings settings, ThreadPool threadPool, TransportService transportService,
+                                 UnicastHostsProvider hostsProvider) {
+        if (getPluginsService().filterPlugins(MockZenPing.TestPlugin.class).isEmpty()) {
```

Comment:
doesn't have to be part of this change, but I wonder if we should use a test only setting for this. Might be simpler.